bundle/manifests/datasciencecluster.opendatahub.io_datascienceclusters.yaml

```diff
@@ -443,6 +443,16 @@ spec:
                         - Removed
                         pattern: ^(Managed|Unmanaged|Force|Removed)$
                         type: string
+                      registriesNamespace:
+                        default: odh-model-registries
+                        description: Namespace for model registries to be installed,
+                          configurable once, defaults to "odh-model-registries"
+                        type: string
+                        x-kubernetes-validations:
+                        - message: RegistriesNamespace is immutable
+                          rule: self == oldSelf
+                    required:
+                    - registriesNamespace
```

Comment:
Hi @etirelli @lburgazzoli @zdtsw this probably is a non issue, but I would like to confirm if adding required fields in DSC is a compatible change  ? Ref: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#on-compatibility .

Reply:
your concern is on 
1. we should uplift API version ?
2. if old DSC CR (esp for modelreg as component) is compatible with this new changes?

Reply:
Right both.

Also for 1. If we should update API version or remove this from required fields

Reply:
Doesn't marking it optional with `omitempty`, but with a default value make the API server always set a value?

Reply:
Based on my testing, K8s considers an empty string value "" valid for this optional field and doesn't use the default value. That's why @zdtsw had to add it to the test DSC create func. 
But, @VaishnaviHire has a point about needing to change API versions if we add a new required property. So, we should avoid that. That's why I'm trying to get it to always have the property, without explicitly setting it to required. There are other properties like KNative Serving Name that also work in a similar fashion.